File: abstract-factory/c-sharp/IDatabaseFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstractFactory
{
    // Oluşturulacak ürünlere ait metotları içerir.
    // Oracle ve MySQL için bunların geriye döndüreceği
    // Nesneler farklı olacaktır.
    interface IDatabaseFactory
    {
        Connection CreateConnection();
        Command CreateCommand();
    }
}

File: abstract-factory/php/IDatabaseFactory.php
<?php

namespace AbstractFactory;

use AbstractFactory\Connection;
use AbstractFactory\Command;

// Oluşturulacak ürünlere ait metotları içerir.
// Oracle ve MySQL için bunların geriye döndüreceği
// Nesneler farklı olacaktır.
interface IDatabaseFactory
{
    function createConnection(): Connection;
    function createCommand(): Command;
}
